src/main/java/com/joyent/manta/monitor/Application.java

```diff
@@ -114,8 +124,17 @@ public static void main(final String[] args) throws InterruptedException {
                 Class<MantaOperationsChain> chainClass =
                         (Class<MantaOperationsChain>)Class.forName(runner.getChainClassName());
                 MantaOperationsChain chain = injector.getInstance(chainClass);
+                Histogram requestPutHistogram = Histogram.build()
+                        .name("manta_monitor_put_request_latency_seconds_"
+                                + chain.getClass().getSimpleName())
+                        .help("Put Request latency in seconds")
+                        .create();
+                // Add the histogram here, to register it later below
+                requestPutHistogramList.add(requestPutHistogram);
+                requestPutHistogramMap.put(chain.getClass().getSimpleName(),
```

Comment:
Let's see if I understand the implementation correctly. We are using a `ConcurrentHashMap` that is shared across all of the running threads that uses a key of the chain name to map to a histogram for a given chain. Each chain thread will always be using the `Histogram` specific to the chain because it is identified by the chain name.

That makes sense to me, but you may want to add a comment to make it easier to understand. I had to spend about 10 minutes thinking it through.